xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/ImageBlock.java

```diff
@@ -121,6 +161,7 @@ public boolean equals(Object obj)
 
             builder.append(getReference(), ((ImageBlock) obj).getReference());
             builder.append(isFreeStandingURI(), ((ImageBlock) obj).isFreeStandingURI());
+            builder.append(getId(), ((ImageBlock) obj).getId());
```

Comment:
`equals/hasCode` can be refactored to use \HashCodeBuilder` and `EqualsBuilder`. See https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HEquals2FHashCodeandToStringimplementations

Reply:
You mean the wrong `obj instanceof ImageBlock` or do you mean that we should use method chaining instead of working with an explicit instance variable?

Reply:
ok, should have checked first. This is simply not using the standard `equals + EqualsBuilder` from Intellij. 
The type of the parameters is tested and the methods stopped early if it is not the case, for instance:
```
@Override
public boolean equals(Object obj)
{
    if (this == obj) {
        return true;
    }

    if (obj == null || getClass() != obj.getClass()) {
        return false;
    }

    ImageBlock that = (ImageBlock) obj;

    return new EqualsBuilder().appendSuper(super.equals(obj))
        .append(this.freestanding, that.freestanding).append(this.reference, that.reference).isEquals();
}
```

Reply:
This is a small perf improvement (and code style improvement as a bunch of repeated casts can be avoided.